src/toil/lib/misc.py

```diff
@@ -118,16 +119,21 @@ def AtomicFileCreate(final_path, keep=False):
                 pass
         raise
 
-def atomic_copy(src_path, dest_path):
+def atomic_copy(src_path, dest_path, executable=False):
     """Copy a file using posix atomic creations semantics."""
     with AtomicFileCreate(dest_path) as dest_path_tmp:
         shutil.copyfile(src_path, dest_path_tmp)
+    if executable:
```

Comment:
I think this should be inside the `with`, which provides the protection that makes this actually atomic. If we leave it out here, we can put the file in place where other people can see it before we fix the permissions.

Inside the `with` we'd have to operate on `dest_path_tmp` instead of `dest_path`; `AtomicFileCreate` moves the one to the other when the `with` ends.